tests/libsubid/04_nss/Makefile

```diff
@@ -1,12 +1,18 @@
 all: test_nss libsubid_zzz.so
 
-test_nss: test_nss.c ../../../lib/nss.c
-	gcc -c -I../../../lib/ -I../../.. -o test_nss.o test_nss.c
-	gcc -o test_nss test_nss.o ../../../lib/.libs/libshadow.a -ldl
+ifdef BASE_TEST_DIR
+basedir := $(BASE_TEST_DIR)
+else
+basedir := $(shell git rev-parse --show-toplevel)
+fi
```

Comment:
This could be rewritten as:

```make
BASE_TEST_DIR ?= $(shell git rev-parse --show-toplevel)
basedir := $(BASE_TEST_DIR)
```

(But feel free to prefer your way.  It's just a suggestion to avoid (explicit) branching.)

Reply:
yeah that does look better, thanks

Reply:
And you could `s/?=/:=/` too, since people rarely pass environment variables to make(1), since it's usually easier to just override the variable as an argument.